app/goal_routes.py

```diff
@@ -0,0 +1,112 @@
+from app import db
+from .models.goal import Goal
+from flask import Blueprint, jsonify, make_response, request, abort
+from app.routes import validate_tasks
+
+
+goals_bp = Blueprint("goals_bp", __name__, url_prefix="/goals")
+
+def validate_goals(goal_id):
+    try:
+        goal_id = int(goal_id)
+    except:
+        abort(make_response({"message":f"Goal {goal_id} invalid."}, 400))
+
+    goal = Goal.query.get(goal_id)
+
+    if not goal:
+        abort(make_response({"message":f"Goal {goal_id} not found."}, 404))
+    
+    return goal
```

Comment:
Nice helper function.  This could probably be shared between goals and routes if you pass the class name in as a parameter. 

```suggestion
def validate_goals(cls, id):
    try:
        id = int(id)
    except:
        abort(make_response({"message":f"{cls} {id} invalid."}, 400))

    instance = cls.query.get(id)

    if not instance:
        abort(make_response({"message":f"{cls} {id} not found."}, 404))
    
    return instance
```